ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,439 @@
+    def _traced_exec_entrypoint_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+        from ddtrace.ext import SpanTypes
+
+        if tracer.current_span() is None:
+            tracer.context_provider.activate(_extract_tracing_context_from_env())
+
+        script_name_match = ENTRY_POINT_REGEX.search(self._entrypoint)
+        if script_name_match:
+            entrypoint_name = f"{script_name_match.group(1)}.py"
+        else:
+            entrypoint_name = os.path.basename(self._entrypoint)
+
+        with tracer.trace(
+            f"exec {entrypoint_name}", service=os.environ.get(RAY_JOB_NAME), span_type=SpanTypes.RAY
+        ) as span:
+            span.set_tag_str(SPAN_KIND, SpanKind.CONSUMER)
+            _inject_ray_span_tags_and_metrics(span)
+
+            return method(self, *args, **kwargs)
+
+    return _traced_exec_entrypoint_method
+
+
+@contextmanager
+def _trace_actor_method(self: Any, method: Callable[..., Any], dd_trace_ctx, *args, **kwargs):
+    context = tracer.context_provider.active()
+    if context is None:
+        context = _TraceContext._extract(dd_trace_ctx)
+
+    with long_running_ray_span(
+        f"{self.__class__.__name__}.{method.__name__}",
+        service=os.environ.get(RAY_JOB_NAME),
+        span_type=SpanTypes.RAY,
+        child_of=context,
+        activate=True,
+    ) as actor_execute_span:
+        set_maybe_big_tag(actor_execute_span, "ray.actor_method.args", args)
+        set_maybe_big_tag(actor_execute_span, "ray.actor_method.kwargs", kwargs)
+
+        yield actor_execute_span
+
+
+def _inject_tracing_actor_method(method: Callable[..., Any]) -> Any:
+    def _traced_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+
+        if _dd_trace_ctx is None and tracer.current_span() is None:
+            return method(self, *args, **kwargs)
+
+        with _trace_actor_method(self, method, _dd_trace_ctx, *args, **kwargs):
+            return method(self, *args, **kwargs)
+
+    return _traced_method
+
+
+def _inject_tracing_async_actor_method(method: Callable[..., Any]) -> Any:
+    async def _traced_async_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+
+        if _dd_trace_ctx is None and tracer.current_span() is None:
+            return await method(self, *args, **kwargs)
+
+        with _trace_actor_method(self, method, _dd_trace_ctx, *args, **kwargs):
+            return await method(self, *args, **kwargs)
+
+    return _traced_async_method
+
+
+def inject_tracing_into_actor_class(wrapped, instance, args, kwargs):
+    cls = wrapped(*args, **kwargs)
+    module_name = str(cls.__module__)
+    class_name = str(cls.__name__)
+
+    # Skip tracing for certain ray modules
+    if module_name.startswith("ray.dag") or module_name.startswith("ray.experimental"):
```

Comment:
Suggestion: Can we define an allowlist instead of denylist. That way we can be intentional about what we wrap. If not, can we at least define the deny list at the top of the file (in a set) for Quick Look up